ratis-server/src/test/java/org/apache/ratis/server/impl/LeaderElectionTests.java

```diff
@@ -96,6 +96,27 @@ public void testChangeLeader() throws Exception {
     cluster.shutdown();
   }
 
+  @Test
+  public void testLostMajorityHeartbeats() throws Exception {
+    runWithNewCluster(3, this::runTestLostMajorityHeartbeats);
+  }
+
+  void runTestLostMajorityHeartbeats(CLUSTER cluster) throws Exception {
+    final RaftServerImpl leader = waitForLeader(cluster);
+    try {
+      isolate(cluster, leader.getId());
+      Thread.sleep(leader.getMaxTimeoutMs());
+      Thread.sleep(leader.getMaxTimeoutMs());
```

Comment:
`Thread.sleep(leader.getMaxTimeoutMs())` duplicated call ?

Reply:
Yes.  The test may fail if it only sleeps once.